src/Esfa.Vacancy.Infrastructure/Services/GetMinimumWagesService.cs

```diff
@@ -41,5 +42,14 @@ public GetAllApprenticeMinimumWagesService(SqlDatabaseService sqlDatabaseService
                 throw new InfrastructureException(e);
             }
         }
+
+        public async Task<WageRange> GetWageRangeAsync(DateTime expectedStartDate)
+        {
+            var ranges = await GetAllWagesAsync();
+
+            return ranges?.FirstOrDefault(range =>
+                range.ValidFrom.Date <= expectedStartDate.Date &&
+                range.ValidTo.Date >= expectedStartDate.Date);
```

Comment:
This method should never be called, but since the interface is shared between cache and actual service I had to put some implementation here. I have changed it to use Single anyway.